src/main/scala/com/linkedin/feathr/offline/transformation/AnchorToDataSourceMapper.scala

```diff
@@ -93,13 +94,19 @@ private[offline] class AnchorToDataSourceMapper(dataPathHandlers: List[DataPathH
       failOnMissingPartition: Boolean): DataFrame = {
 
     val dataLoaderHandlers: List[DataLoaderHandler] = dataPathHandlers.map(_.dataLoaderHandler)
-    val pathChecker = PathChecker(ss)
-    val pathAnalyzer = new TimeBasedHdfsPathAnalyzer(pathChecker, dataLoaderHandlers)
```

Comment:
@blee1234 still, this line does **not** work with not only Jdbc source but also Kafka, 
Please **DO NOT** assume that every data source is on HDFS, it is **NOT TRUE**.

Reply:
@windoze Suggest to add comment in the code to explain this intention. Will be easier for folks to understand.

Reply:
> 

I did add comments in my code, this message is my comment on original code, seems he forgot that a data source could be something other than HDFS even before this PR.

Reply:
@windoze sorry I'm confused on your comment. for line 97, my change simply passes dataLoaderHandlers as a parameter to the the TimeBasedHdfsPathAnalyzer. 

If you're looking at who added the TimeBasedHdfsPathAnalyzer in the first place, you'll have to go through the git history.

Reply:
> 

Sorry for looking at the wrong guy, I was just wondering, why the code is fully of assumptions that the data source comes solely from HDFS and makes even `local override` doesn't fully work as expected.
I'll start a major cleanup soon, and wish you can provide some help.